source/common/common/logger.cc

```diff
@@ -103,13 +107,22 @@ Context::~Context() {
   }
 }
 
-void Context::activate() {
+void Context::activate(LoggerMode mode) {
   Registry::getSink()->setLock(lock_);
   Registry::getSink()->set_should_escape(should_escape_);
   Registry::setLogLevel(log_level_);
   Registry::setLogFormat(log_format_);
+
+  if (mode == LoggerMode::Fancy) {
```

Comment:
I am not sure this is the right place to set your default log level. class Context is used to override logging behavior when it is created and restore the previous behavior when it is destructed.

But getFancyDefaultLevel is only used once for each log key (file name), meaning the Context class does not affect fancy log's behavior once fancy log is initialized.

I think the better way is to provide a pair of function in FancyContext that iterates through the loggers and change their log level and format.

Reply:
The function is added in FancyContext to enable later update of default level/format of different Context, and it's called in activate() now. But I think it's still necessary to set the default level in Context() as the map isn't complete when the 1st Context is initialized.

Reply:
Maybe make fancy_default_level_ and fancy_log_format_ non-static member? You can access them via the current_context_ global var. I think this makes it clearer about the life span of the fancy_default_level_ and fancy_log_format_.

Reply:
Seems making them non-static needs another variable to represent the previous default level. Currently I set default before setting fancy_default_level so that I can compare new level with the old one.